docs/tutorials/semantic_search/generate_embeddings_for_arrays_of_object.md

```diff
@@ -0,0 +1,299 @@
+
+# Steps
+
+## 1. Create embedding model
+
+We will use [Bedrock Titan Embedding model](https://docs.aws.amazon.com/bedrock/latest/userguide/titan-embedding-models.html) in this tutorial. 
+
+- If you are using AWS managed OpenSearch service, you can use this [python notebook](https://github.com/opensearch-project/ml-commons/blob/2.x/docs/tutorials/aws/AIConnectorHelper.ipynb) to create Bedrock Embedding Model easily. Search `1. Create Connector of Bedrock Embedding Model` on the page. 
+Or you can manually create connector following this [tutorial](https://github.com/opensearch-project/ml-commons/blob/2.x/docs/tutorials/aws/semantic_search_with_bedrock_titan_embedding_model.md).
+
+- If you are using self-managed OpenSearch, you can follow this [blueprint](https://github.com/opensearch-project/ml-commons/blob/2.x/docs/remote_inference_blueprints/bedrock_connector_titan_embedding_blueprint.md). 
+
+Use the model ID from the response to test predict API:
+```
+POST /_plugins/_ml/models/your_embedding_model_id/_predict
+{
+    "parameters": {
+        "inputText": "hello world"
+    }
+}
+```
+Sample response:
+
+```
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "sentence_embedding",
+          "data_type": "FLOAT32",
+          "shape": [ 1536 ],
+          "data": [0.7265625, -0.0703125, 0.34765625, ...]
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
+```
+
+## 2. Create ingest pipeline
+
+### 2.1 Create test index
+```
+PUT my_books
+{
+  "settings" : {
+      "index.knn" : "true",
+      "default_pipeline": "bedrock_embedding_foreach_pipeline"
+  },
+  "mappings": {
+    "properties": {
+      "books": {
+        "type": "nested",
+        "properties": {
+          "title_embedding": {
+            "type": "knn_vector",
+            "dimension": 1536
+          },
+          "title": {
+            "type": "text"
+          },
+          "description": {
+            "type": "text"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### 2.2 Create ingest pipeline
+
+Create sub-pipeline to generate embedding for one item in the array.
+
+This pipeline contains 3 processors
+- set processor: The `text_embedding` processor is unable to identify "_ingest._value.title". You need to copy "_ingest._value.title" to a temporary field for text_embedding to process it.
+- text_embedding processor: convert value of the temporary field to embedding
+- remove processor: remove temporary field
+```
+PUT _ingest/pipeline/bedrock_embedding_pipeline
+{
+  "processors": [
+    {
+      "set": {
+        "field": "title_tmp",
+        "value": "{{_ingest._value.title}}"
+      }
+    },
+    {
+      "text_embedding": {
+        "model_id": your_embedding_model_id,
+        "field_map": {
+          "title_tmp": "_ingest._value.title_embedding"
+        }
+      }
+    },
+    {
+      "remove": {
+        "field": "title_tmp"
+      }
+    }
+  ]
+}
+```
+
+Create pipeline with foreach processor:
```

Comment:
May be we can add a link for `foreach` processor to explain what does this processor do?

Reply:
Unfortunately, OpenSearch doesn't have foreach processor doc. There is one issue tracking the gap https://github.com/opensearch-project/documentation-website/issues/4647